framework/packages/gov-ownable/src/lib.rs

```diff
@@ -0,0 +1,565 @@
+/// it's not the given address.
+/// Return Err if fails to load ownership info from storage.
+pub fn is_owner(store: &dyn Storage, querier: &QuerierWrapper, addr: &Addr) -> StdResult<bool> {
+    let ownership = OWNERSHIP.load(store)?;
+
+    if let Some(owner) = ownership.owner.owner_address(querier) {
+        if *addr == owner {
+            return Ok(true);
+        }
+    }
+
+    Ok(false)
+}
+
+/// Assert that an account is the contract's current owner.
+pub fn assert_owner(
+    store: &dyn Storage,
+    querier: &QuerierWrapper,
+    sender: &Addr,
+) -> Result<(), GovOwnershipError> {
+    let ownership = OWNERSHIP.load(store)?;
+    check_owner(querier, &ownership, sender)
+}
+
+/// Assert that an account is the contract's current owner.
+fn check_owner(
+    querier: &QuerierWrapper,
+    ownership: &Ownership<Addr>,
+    sender: &Addr,
+) -> Result<(), GovOwnershipError> {
+    // the contract must have an owner
+    let Some(current_owner) = &ownership.owner.owner_address(querier) else {
+        return Err(GovOwnershipError::NoOwner);
+    };
+
+    // the sender must be the current owner
+    if sender != current_owner {
+        return Err(GovOwnershipError::NotOwner);
+    }
+
+    Ok(())
+}
+
+/// Update the contract's ownership info based on the given action.
+/// Return the updated ownership.
+pub fn update_ownership(
+    deps: DepsMut,
+    block: &BlockInfo,
+    sender: &Addr,
+    version_control: Addr,
+    action: GovAction,
+) -> Result<Ownership<Addr>, GovOwnershipError> {
+    match action {
+        GovAction::TransferOwnership { new_owner, expiry } => {
+            transfer_ownership(deps, sender, new_owner, version_control, expiry)
+        }
+        GovAction::AcceptOwnership => accept_ownership(deps.storage, &deps.querier, block, sender),
+        GovAction::RenounceOwnership => renounce_ownership(deps.storage, &deps.querier, sender),
+    }
+}
+
+/// Get the current ownership value.
+pub fn get_ownership(storage: &dyn Storage) -> StdResult<Ownership<Addr>> {
+    OWNERSHIP.load(storage)
+}
+
+pub fn query_ownership<Q: CustomQuery>(
+    querier: &QuerierWrapper<Q>,
+    remote_contract: Addr,
+) -> StdResult<Ownership<Addr>> {
+    OWNERSHIP.query(querier, remote_contract)
+}
+
+/// Propose to transfer the contract's ownership to the given address, with an
+/// optional deadline.
+fn transfer_ownership(
+    deps: DepsMut,
+    sender: &Addr,
+    new_owner: GovernanceDetails<String>,
+    version_control: Addr,
+    expiry: Option<Expiration>,
+) -> Result<Ownership<Addr>, GovOwnershipError> {
+    let new_owner = new_owner.verify(deps.as_ref(), version_control)?;
+
+    if let GovernanceDetails::Renounced {} = new_owner {
+        return Err(GovOwnershipError::TransferToRenounced {});
+    }
+
+    OWNERSHIP.update(deps.storage, |ownership| {
+        // the contract must have an owner
+        check_owner(&deps.querier, &ownership, sender)?;
+
```

Comment:
What I think we should add here is a check for if the current owner is of type NFT. If so, we should prevent the ownership transfer. Otherwise any NFT holder can transfer the account away from the NFT (which is probably not something the NFT creators expect the user to be able to do!).